### PR TITLE
added to catalog layout for consistent levels

### PR DESCRIPTION
**Pull request by cholmes**
**Related Issue(s):** #949 

**Proposed Changes:**

1. Added best practice to catalog layout for consistent levels

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: best-practices.md [hunk 1]
```diff
@@ -274,6 +275,9 @@ ended up doing. Following these recommendations makes for more legible catalogs.
 5. Items should be stored in subdirectories of their parent catalog. 
 This means that each item and its assets are contained in a unique subdirectory.
 6. Limit the number of items in a catalog or sub-catalog, grouping / partitioning as relevant to the dataset.
+7. In a nested hierarchy of catalogs and collections make the different levels 'homogeneous' - all catalogs or all collections.
```
Comment (m-mohr) [inline best-practices.md:278]: That sounds a bit like there should be only one level of collections, although it should also be fine to have e.g. all levels being collections.

Comment (cholmes) [reply best-practices.md:278]: Cool, I'll try again. I was thinking about that as I was writing it but didn't manage to find a way to succinctly get it across. This was weirdly more challenging to explain than I was expecting.

**Comment by cholmes** [comment]
Ok, new attempt is up.

File: best-practices.md [hunk 2]
```diff
@@ -404,6 +405,8 @@ ended up doing. Following these recommendations makes for more legible catalogs.
 5. Items should be stored in subdirectories of their parent Catalog. 
 This means that each Item and its assets are contained in a unique subdirectory.
 6. Limit the number of Items in a Catalog or sub-Catalog, grouping / partitioning as relevant to the dataset.
+7. Use structural elements (Catalog and Collection) consistently across each 'level' of your hierarchy. If levels 2 and 4
+have are both only Collections then don't add a Catalog there (and vice versa), or else change every element at that level.
```
Comment (lossyrob) [inline best-practices.md:409]:
```suggestion
7. Use structural elements (Catalog and Collection) consistently across each 'level' of your hierarchy. For example, if levels 2 and 4 of the hierarchy only contain Collections, 
don't add a Catalog at levels 2 and 4.
```

**Review by lossyrob — APPROVED** [review]
Suggested some wording changes to make it a bit more clear.

**Review by m-mohr — APPROVED** [review]

**Pull request merged**